src/utils.js

```diff
@@ -84,7 +84,8 @@ export function debounce (fn) {
 export function getClosestIgnoredElement (element) {
   var parent = element;
 
-  while (parent && parent !== document && !(parent instanceof DocumentFragment)) {
+  // e.g. document doesn't have a function hasAttribute; no need to go further up
+  while (parent && typeof parent.hasAttribute === 'function' && !(parent instanceof DocumentFragment)) {
```

Comment:
> Not only document element doesn't have hasAttribute method

That's why I suggested it.

> constructor check is way more bulletproof and explicit.

That's true. So maybe we should re-evaluate exactly what we want to get from this check. In my mind we want to ensure that we can call `hasAttribute` on it, right? There's a few options here:

1. Check `instanceof` on both `HTMLDocument` and `DocumentFragment`
2. Check `instanceof` on just `Element` (since that's the interface that implements `hasAttribute`)
3. Simply check if the `parent` has `hasAttribute`

I don't mind option 3. I don't see any real benefits to being super explicit here. If it has `hasAttribute` then we should be able to call it, right?

Option 2 is probably a good balance between the two. Concise and we have two checks instead of one.